### PR TITLE
Set up auto to automate releases

### LINKED ISSUE #73: auto-mate releases
@jwodder would you be so kind?

### PR DESCRIPTION
**Pull request by jwodder**
Closes #73.

This PR also needs to be accompanied by the following changes:
- [x] Create the necessary labels for `auto` by running `GH_TOKEN=... auto create-labels` in a copy of this repository containing `.autorc`
- [x] All PRs merged since the previous release should receive appropriate labels
- [x] An auth token for uploading assets for the `pyout` project to PyPI must be saved as a secret named "`PYPI_TOKEN`"

### CODE REVIEW

**Comment by vsoch** [comment]
In case it helps I use this workflow to release to pypi, after logging in and creating a project specific token. https://github.com/singularityhub/singularity-hpc/blob/main/.github/workflows/release.yaml.

I still like control of _when_ the release happens, but with automated release notes that is just a few clicks!

**Comment by yarikoptic** [comment]
> I still like control of _when_ the release happens, but with automated release notes that is just a few clicks!

if I got the accent on _when_ correctly, we will still have that control -- releases would be triggered by merging PRs with `release` label.

File: .autorc [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    "noVersionPrefix": false,
+    "plugins": [
+        "git-tag",
+        [
+            "exec",
+            {
+                "afterChangelog": "bump2version \"$(printf '%s\n' \"$ARG_0\" | jq -r .bump)\"",
+                "afterRelease": "python -m build && twine upload dist/*"
+            }
+        ]
```
Comment (yarikoptic) [inline .autorc]:
```suggestion
        ],
        "released"
```

I like it too much I guess ;)

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: Auto-release on PR merge
```
Comment (yarikoptic) [inline .github/workflows/release.yml:1]: could it be also used/allowed to be triggered manually among actions on current shape of master branch?

**Comment by yarikoptic** [comment]
@vsoch - would you mind adding me (yarikoptic) to maintainers of tributors on PyPI?

File: .github/workflows/release.yml (hunk 2 above)
Comment (jwodder) [reply .github/workflows/release.yml:1]: @yarikoptic What do you mean by "on current shape of master branch"?

Comment (yarikoptic) [reply .github/workflows/release.yml:1]: Just that we could trigger release of current master version without PR

Comment (jwodder) [reply .github/workflows/release.yml:1]: f4b1a56 should do it.

**Comment by vsoch** [comment]
Are you sure we don't need a video call before to verify you are who you are? How do I know someone hasn't hijacked yarikoptic's account? :thinking: 

Just kidding, that would be ridiculous :)

Done.

**Comment by yarikoptic** [comment]
It all depends, it all depends... Thank you!

**Comment by yarikoptic** [comment]
oh, need to get back to this one too

**Comment by yarikoptic** [comment]
@jwodder I do not see `release` and only see `skip-release`. Does it mean that we would release after each PR merge?

**Comment by jwodder** [comment]
@yarikoptic The lack of a `release` label is a side-effect of f4b1a56b41bb3eac2971e12ed11988abbc8d3734.  The label will have to be created manually.

**Comment by yarikoptic** [comment]
gotcha -- thanks, made a label

**Pull request merged**